cg/services/delivery_message/utils.py

```diff
@@ -24,20 +24,24 @@
     MicrosaltMwxMessage,
 )
 from cg.store.models import Case, Sample
+from cg.store.store import Store
 
 
-def get_message(cases: list[Case]) -> str:
-    message_strategy: DeliveryMessage = get_message_strategy(cases[0])
+def get_message(cases: list[Case], store: Store) -> str:
+    message_strategy: DeliveryMessage = get_message_strategy(cases[0], store)
     return message_strategy.create_message(cases)
 
 
-def get_message_strategy(case: Case) -> DeliveryMessage:
+def get_message_strategy(case: Case, store: Store) -> DeliveryMessage:
```

Comment:
An alternative to adding a store here and RNAMessage classes is adding the crud function as a property of `Case`.

Reply:
Then I prefer passing the store here. Although perhaps not elegant, I think that has more to do with the way we link RNA and DNA cases than code design.

Reply:
I like the store here

Reply:
Having it on the case model in the database will introduce more business logic to the database layer. We will run into issues with separation of concerns. Also it adds more pressure on the case model in regards to testing- you always need to make sure that you get the right message strategy back from the case model. So it will be harder to test things where the model is involved.

We should strive to keep the database layer only to manage and persist data.

Reply:
It stays here then!